### ISSUE TITLE
the version of @typescript-eslint/typescript-estree needs to be selected

### ISSUE THREAD

**Issue by fencer-yd** [typescript-eslint/typescript-eslint#5303]
### Before You File a Bug Report Please Confirm You Have Done The Following...

- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [searched for related issues](https://github.com/typescript-eslint/typescript-eslint/issues?q=is%3Aissue+label%3Abug) and found none that matched my issue.
- [X] I have [read the FAQ](https://typescript-eslint.io/docs/linting/troubleshooting) and my problem is not listed.

### Relevant Package

typescript-estree

### Repro Code

```typescript
$ yarn install
# or
$ npm install
```

### ESLint Config

```javascript
module.exports = {
  parser: "@typescript-eslint/parser",
  rules: {
    "@typescript-eslint/<rule-name>": ["error", ...<options>],
  },
};
```

### tsconfig

```jsonc
{
  "compilerOptions": {
    // ...
  }
}
```

### Expected Result

Can execute yarn install/npm install normally.

### Actual Result

There is no stable 5.30.4 version for @typescript-eslint/typescript-estree. As a result, the version of @typescript-eslint/typescript-estree needs to be selected every time in yarn install/npm install.

### Versions

| package                                | version |
| -------------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin`     | `X.Y.Z` |
| `@typescript-eslint/parser`            | `X.Y.Z` |
| `@typescript-eslint/scope-manager`     | `X.Y.Z` |
| `@typescript-eslint/typescript-estree` | `5.30.4` |
| `@typescript-eslint/type-utils`        | `X.Y.Z` |
| `@typescript-eslint/utils`             | `X.Y.Z` |
| `TypeScript`                           | `X.Y.Z` |
| `ESLint`                               | `X.Y.Z` |
| `node`                                 | `X.Y.Z` |

**Comment by Josh-Cena** [comment]
@fencer-yd this may help: https://gist.github.com/hzoo/df94af2d2ec48696e683b772c775aa67

TL;DR: if you are using a proxy, try clearing your proxy cache; if you are using a mirror registry, check if that registry has synced the latest version (https://npmmirror.com/package/%40typescript-eslint%2Ftypescript-estree is available)

**Comment by fencer-yd** [comment]
> @fencer-yd this may help: https://gist.github.com/hzoo/df94af2d2ec48696e683b772c775aa67
> 
> TL;DR: if you are using a proxy, try clearing your proxy cache; if you are using a mirror registry, check if that registry has synced the latest version (https://npmmirror.com/package/%40typescript-eslint%2Ftypescript-estree is available)

Thank you for your reply, I think it is very likely that the mirror proxy problem. I see 5.30.4 has been released 20 hours ago.
